pkg/store/instrumented/instrumented.go

```diff
@@ -0,0 +1,53 @@
+package instrumented
+
+import (
+	"context"
+
+	"github.com/openshift/telemeter/pkg/metricfamily"
+	"github.com/openshift/telemeter/pkg/store"
+	"github.com/prometheus/client_golang/prometheus"
+	clientmodel "github.com/prometheus/client_model/go"
+)
+
+var (
+	metricSamples = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "telemeter_server_samples",
+		Help: "Tracks the number of samples processed by this server.",
+	}, []string{"phase"})
+)
+
+func init() {
+	prometheus.MustRegister(metricSamples)
+}
+
+type instrumented struct {
+	target store.Store
+	gauge  prometheus.Gauge
```

Comment:
This is should be a counter, not a gauge. The number of written metrics should never need to be reset (and in fact it is never reset in this code), only incremented monotonically.

Reply:
agreed, this refactoring didn't change the metrics type, but we could do it here. I'll add the change to a counter in this PR.